### PR TITLE
Added object to single section selection delegate callback

### LINKED ISSUE #396: IGListSingleSectionControllerDelegate selection function should include object
Right now to get the object from the `selectionDelegate`, you have to:

```swift
func didSelect(_ sectionController: IGListSingleSectionController) {
  let object = adapter.object(for: sectionController)
  // do something
}
```

When `IGListSingleSectionController` already has [the object](https://github.com/Instagram/IGListKit/blob/master/Source/IGListSingleSectionController.m#L23), this lookup is wasteful.

I propose we change the delegate method to:

```objc
- (void)didSelectSingleSectionController:(IGListSingleSectionController *)sectionController 
  withObject:(id)object;
```

- What do you think about the naming? "withObject" or maybe just "object"?
- Should we shorten the method to just `didSelectSectionController`?

### PR DESCRIPTION
**Pull request by Sherlouk**
## Changes in this pull request

- Updated selectionDelegate to pass through the model object
- Changelog entry under 3.0.0 as is breaking change

Try saying single section selection 10 times quick
Dabbing my left toe into some ObjC, so apologies if I violate years of rules

- [ ] Update documentation internally

Closes #396 

## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by jessesquires** [comment]
Thanks @Sherlouk ! 💯 Looks good to me.

This is a 3.0 breaking change though, I opened #398 to discuss.

cc @rnystrom

File: Source/IGListSingleSectionController.h [hunk 1]
```diff
@@ -45,8 +45,10 @@ typedef CGSize (^IGListSingleSectionCellSizeBlock)(id item, id<IGListCollectionC
  Tells the delegate that the section controller was selected.
 
  @param sectionController The section controller that was selected.
+ @param object            The model for the given section
```
Comment (jessesquires) [inline Source/IGListSingleSectionController.h]: super nit: need a period `.` 😄

**Comment by Sherlouk** [comment]
Travis/Tests are being a tad flakey, builds fine

As per discussion should be fine to merge into `master`

**Comment by jessesquires** [comment]
Pushed a changelog update here, per #398

**Comment by jessesquires** [comment]
@Sherlouk - question -- can you push your own branches to `Insta/IGListKit` instead using your fork?

That *should* be possible now. If so, let's do that. That way, we can all checkout any branch, push updates/fixes, etc.

**Comment by Sherlouk** [comment]
@jesse_squires as per discussion in #399 I don't have write access yet but it's being worked on

Cc @rnystrom

**Comment by Sherlouk** [comment]
Conflict resolution feature on GitHub is pretty cool

**Comment by jessesquires** [comment]
oof. wanted to wait on merging this.

this is the only 3.0 change so far right? i'll see if i can fix up the branches

**Comment by jessesquires** [comment]
ok. got everything sync'd up correctly on `stable` locally. now having GH permissions issues. will push once that's resolved

**Comment by rnystrom** [comment]
@jessesquires ah sorry my bad, didn't see this needed to wait. Will report back on protected branch.

**Pull request closed without merging**